projects/client/RabbitMQ.Client/src/client/impl/AutorecoveringConnection.cs

```diff
@@ -76,8 +76,7 @@ public class AutorecoveringConnection : IConnection, IRecoverable
         protected ConcurrentDictionary<RecordedBinding, byte> m_recordedBindings =
             new ConcurrentDictionary<RecordedBinding, byte>();
 
-        protected List<EventHandler<ConnectionBlockedEventArgs>> m_recordedBlockedEventHandlers =
-            new List<EventHandler<ConnectionBlockedEventArgs>>();
+        protected EventHandler<ConnectionBlockedEventArgs> m_recordedBlockedEventHandlers;
```

Comment:
Changing the type of protected member would be a breaking change to any derived types. `AutorecoveringConnection` is a public type, so if you're strictly avoiding breaks, this shouldn't be changed.

Reply:
I am not aware of any projects that would derive from this class. Does NServiceBus do it?

Reply:
No, we don't. I'm guessing the odds are low that anyone does, but those types are `protected`, which means that someone could observe the change, and you'd break them, though.

Reply:
these can all be made private. Now that work on 6.0 has started I would happily accept PRs that hides fields and data that ought to be internal.